src/k2/cpo/service/CPOService.cpp

```diff
@@ -383,6 +411,44 @@ void CPOService::_assignCollection(dto::Collection& collection) {
         }));
 }
 
+seastar::future<> CPOService::_offloadCollection(dto::Collection& collection) {
```

Comment:
you don't have any retries here(which is fine) but we should make sure we can retry from the outside if things time out/fail.

I think to achieve that we need:
- 404_not_found response from node if the collection has already been offloaded there
- handling of 404 here as a non-error condition (along with the 200 ok)
- we should return any other error status out and only clean up the cpo state if everything succeeded
- we should be able to mark the cpo state as "deleted" when we receive a drop request. That way we can still have the metadata around but we know not to return it to clients. The metadata should be here until all partitions ACK the drop. If any fail, we leave it untouched and let the client retry if they want